tests/integration/targets/dynamodb_table/tasks/main.yml

```diff
@@ -0,0 +1,207 @@
+      environment:
+        EC2_REGION: "{{ ec2_region }}"
+        EC2_ACCESS_KEY: bogus_access_key
+        EC2_SECRET_KEY: bogus_secret_key
+      register: result
+      ignore_errors: yes
+
+    - name: assert dynamodb_table with valid ec2_url
+      assert:
+        that:
+          - result is failed
+          - >
+            "Authentication failure" in result.msg
+            or "token included in the request is invalid" in result.msg
+
+    # ============================================================
+    - name: test credential parameters
+      dynamodb_table:
+        name: "{{ table_name }}"
+        hash_key_name: myhashkey
+        ec2_region: "{{ ec2_region }}"
+        ec2_access_key: bogus_access_key
+        ec2_secret_key: bogus_secret_key
+        security_token: "{{ security_token | d(omit) }}"
+      register: result
+      ignore_errors: yes
+    - name: assert credential parameters
+      assert:
+        that:
+          - result is failed
+          - >
+            "Authentication failure" in result.msg
+            or "token included in the request is invalid" in result.msg
+
+- name: dynamodb table integration tests
+  module_defaults:
+    dynamodb_table:
+      aws_access_key: "{{ aws_access_key }}"
+      aws_secret_key: "{{ aws_secret_key }}"
+      region: "{{ aws_region }}"
+      security_token: "{{ security_token | d(omit) }}"
+  block:
+    # ============================================================
+    - name: create table
+      dynamodb_table:
+        name: "{{ table_name }}"
+        hash_key_name: myhashkey
+        range_key_name: myrangekey
+        state: present
+        indexes:
+          - name: NamedIndex
+            type: global_all
+            hash_key_name: myhashkey
+          - name: NamedIndexV2
+            type: global_all
+            hash_key_name: myhashkey
+            range_key_name: myrangekey
+          - name: NamedLocalIndex
+            type: keys_only
+            hash_key_name: myhashkey
+            range_key_name: mylocalrangekey
+        tags:
+          First: one
+          Second: two
+      register: result
+      until:
+        - result | json_query('global_secondary_indexes[?IndexName==`NamedIndexV2`].IndexStatus') == ['ACTIVE']
+        - result | json_query('global_secondary_indexes[?IndexName==`NamedIndex`].IndexStatus') == ['ACTIVE']
+      retries: 6
+      delay: 10
+
+    - name: update index on the existing table
+      dynamodb_table:
+        name: "{{ table_name }}"
+        hash_key_name: myhashkey
+        indexes:
+          - name: NamedIndex
+            type: global_all
+            hash_key_name: myhashkey
+          - name: NamedIndexV2
+            type: global_all
+            hash_key_name: myhashkey
+            range_key_name: myrangekey
+            read_capacity: 10
+            write_capacity: 10
+      register: result
+      until:
+        - result | json_query('indexes[?name==`NamedIndexV2`].read_capacity') == [10]
```

Comment:
The `json_query` filter has moved to a collection, so these will need to be `community.general.json_query`.